api/routes/books.js

```diff
@@ -25,54 +27,125 @@ router.get('/:id', async (req, res, next) => {
 })
 
 // You should only be able to create a book if the user is an admin
+// Nested try/catch here
 router.post('/', async (req, res, next) => {
   const status = 200
   try {
-    const book = await Book.create(req.body)
-    if (!book) throw new Error(`Request body failed: ${JSON.stringify(req.body)}`)
-    
-    const response = await Book.findById(book._id).select('-__v')
-    res.json({ status, response })
+    const token = req.headers.authorization.split('Bearer ')[1]
+    //make sure token exists
+    if (!token) throw new Error (`You are not authorized`)
+    const payload = jsonwebtoken.verify(token, SECRET_KEY)
+    const user = await User.findOne({ _id: payload.id }).select('-__v -password')
+    if (!user || user.admin !== true) throw new Error (`You are not authorized`)
+
+    try {
+      //need to insert stuff here 
+      const book = await Book.create(req.body)
+      if (!book) throw new Error(`Request body failed: ${JSON.stringify(req.body)}`)
+      
+      const response = await Book.findById(book._id).select('-__v')
+      res.json({ status, response })
+    } catch (e) {
+      console.error(e)
+      const message = 'Failure to create. Please check request body and try again.'
+      const error = new Error(message)
+      error.status = 400
+      next(error)
+    }
+
   } catch (e) {
-    console.error(e)
-    const message = 'Failure to create. Please check request body and try again.'
-    const error = new Error(message)
-    error.status = 400
-    next(error)
+    e.status = 401
+    next(e)
   }
 })
 
 // You should only be able to reserve a book if a user is logged in
 router.patch('/:id/reserve', async (req, res, next) => {
   const { id } = req.params
+  const status = 200
+  const token = req.headers.authorization.split('Bearer ')[1]
+  //make sure token exists
+  if (!token) {
+    const error = new Error (`You are not authorized`)
+    error.status = 401
+    return next(error)
+  }
+  //try/catch
   try {
+    const payload = jsonwebtoken.verify(token, SECRET_KEY)
```

Comment:
This is what you want to use for whether or not the person is authorized. Just having a token is not enough.